### DISCUSSION TITLE
Support for HugesPages RAM

### DISCUSSION THREAD

**Discussion by fantomfp** [processone/ejabberd#4046] [Ideas]
Is it possible to implement support for HugesPages on GNU/Linux platform? Windows calls them Large Pages.

It should increase performances of ejabberd, if enabled on the host machine.

Summary, HugesPages is a kernel feature which let the software (ejabberd) to use the RAM on its own way. For example mariadb has the same feature ( https://mariadb.com/kb/en/mariadb-memory-allocation/#huge-pages ).

I don't know if the erlang programing language permit to do that. Maybe there is other considerations to take in mind.

Ref: https://wiki.debian.org/Hugepages
https://linuxconfig.org/how-to-enable-hugepages-on-linux

**Comment by licaon-kter** [comment]
The mariadb link ends up with _"Overall performance gain: A few percent. Yawn. Too much hassle for too little benefit."_ :) :)

**Reply by fantomfp** [reply]
> The mariadb link ends up with _"Overall performance gain: A few percent. Yawn. Too much hassle for too little benefit."_ :) :)

I would say I did it and saw mariadb server ~20% faster, but that's only my opinion (user oriented) :D

That was not such a pain to configure the system...

**Comment by prefiks** [comment]
This will require support for this erlang vm, and i don't see any flag that can trigger this.

**Comment by fantomfp** [comment]
This thread of mails was not easy to find, because they wrote the word "HugePages" mispelled:

https://erlang.org/pipermail/erlang-questions/2020-May/099495.html
